### PR TITLE
Emit event in watch of "checkedData" and click event

### PR DESCRIPTION
**Pull request by NFSS10**
| - | - |
| --- | --- |
| Issue | - |
| Decisions | • Emit event in watch of "checkedData"<br>• Emit click event when checkbox is clicked |
| Animated GIF | - |

### CODE REVIEW

File: vue/components/ui/atoms/checkbox/checkbox.vue [hunk 1]
```diff
@@ -178,7 +179,6 @@ export const Checkbox = {
             if (this.disabled) return;
 
             this.checkedData = !this.checkedData;
-            this.$emit("update:checked", this.checkedData);
```
Comment (gcandal) [inline vue/components/ui/atoms/checkbox/checkbox.vue]: 👍

**Comment by gcandal** [comment]
@NFSS10 we should stick with a boolean `checked` and just alternate the `icon` for representing "partial" or whatever it is. Any reason for not doing so?

**Comment by NFSS10** [comment]
> @NFSS10 we should stick with a boolean `checked` and just alternate the `icon` for representing "partial" or whatever it is. Any reason for not doing so?

Yes, as stated in the issue description. As it was before this change, to set the checkbox like this  imagem we need to set the props `icon = "minus"` and `checked = true`. This works well on its own but has a side effect when using the checkbox as a global checkbox (see issue description) because when setting `checked = true`, it should select all of the other checkboxes, so there is a conflict here.

**Comment by gcandal** [comment]
@NFSS10 let's talk more about this, please remind me tomorrow

**Pull request merged**